contracts/cw721-base/src/contract.rs

```diff
@@ -507,25 +507,22 @@ fn query_tokens(
     limit: Option<u32>,
 ) -> StdResult<TokensResponse> {
     let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
-    let start_addr = maybe_addr(deps.api, start_after)?;
-    let start = start_addr.map(|addr| Bound::exclusive(addr.as_ref()));
+    let start = start_after.map(Bound::exclusive);
 
     let owner_addr = deps.api.addr_validate(&owner)?;
     let res: Result<Vec<_>, _> = tokens()
         .idx
         .owner
-        .pks(
-            deps.storage,
-            Vec::from(owner_addr.as_ref()),
-            start,
-            None,
-            Order::Ascending,
-        )
+        .prefix(Vec::from(owner_addr.as_ref()))
+        .range(deps.storage, start, None, Order::Ascending)
         .take(limit)
         .collect();
-    let pks = res?;
+    let tokens = res?;
 
-    let res: Result<Vec<_>, _> = pks.iter().map(|v| String::from_utf8(v.to_vec())).collect();
+    let res: Result<Vec<_>, _> = tokens
+        .iter()
+        .map(|v| String::from_utf8(v.0.clone()))
```

Comment:
And we throw away the data here, just using the pk.

I understand why you show `range` for the example, but since we only want IDs, it is rather inefficient to do it this way
(maybe show both alternatives in the doc and not modify the code?)

Reply:
No problem. I Just wanted to document  some actual code, yes.

Reply:
I think a good idea would be to move `pks()` to `Prefix`. So that `prefix()` can be used with `pks()` the same way we use it with `range()`. And then we write a `pks()` helper in the base map for the empty prefix case.